### PR TITLE
 drivers: gpio: Add support for daisy-chained irqs in mcux driver

### LINKED ISSUE #13122: build for KW40z, KW41z fails to generate isr_tables
**Describe the bug**
Building an example for a frdm_kw41z results in a failure in generating isr_tables when gpio portb is enabled:
```
[ 97%] Generating isr_tables.c
gen_isr_tables.py: multiple registrations at table_index 31 for irq 31 (0x1f)
Traceback (most recent call last):
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 308, in <module>
    main()
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 300, in main
    error("multiple registrations at table_index %d for irq %d (0x%x)" % (table_index, irq, irq))
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 35, in error
    raise Exception()
Exception
make[2]: *** [zephyr/CMakeFiles/kernel_elf.dir/build.make:63: zephyr/isr_tables.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:314: zephyr/CMakeFiles/kernel_elf.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

**To Reproduce**
1. enable GPIO_MCUX_PORTB in boards/arm/frdm_kw41z/Kconfig.defconfig
2. build e.g. hello_world example
3. cmake -DBOARD=frdm_kw41z
4. make
5. See error

**Impact**
Examples for frdm_kw41z or boards based on kw40z and kw41z will fail in building when all gpios are activated

**Screenshots or console output**
in debug mode, I'm getting this:
```
[ 97%] Generating isr_tables.c
gen_isr_tables.py: (32, 0)
gen_isr_tables.py: Configured interrupt routing
gen_isr_tables.py: handler    irq flags param
gen_isr_tables.py: --------------------------
gen_isr_tables.py: 0x1957     31  0     0x20000ed0
gen_isr_tables.py: 0x1957     31  0     0x20000edc
gen_isr_tables.py: 0x1957     30  0     0x20000ee8
gen_isr_tables.py: offset is 0
gen_isr_tables.py: num_vectors is 32
gen_isr_tables.py: multiple registrations at table_index 31 for irq 31 (0x1f)
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: 0f3bd0f43bf5bbc559303dc91d4fc3aacb58281e

**Additional context**
This check was recently introduced by #11273 by @andrewboie, issue is similar to k64 in #11275.
possible fix: Assign a different interrupt number in nxp_kw40z.dtsi and nxp_kw41z.dtsi.
How were the numbers 30/31 selected? Which number to select instead of 31?

### PR DESCRIPTION
**Pull request by MaureenHelm**
Typically we manage driver irqs on a per-instance basis, however the
kw40z and kw41z have a shared irq vector for portb and portc instances.
Adds a new optional device tree binding and driver support to allow
daisy chaining the mcux gpio irq handlers together.

Fixes #13122

### CODE REVIEW

File: drivers/gpio/gpio_mcux.c [hunk 1]
```diff
@@ -220,12 +229,31 @@ static void gpio_mcux_port_isr(void *arg)
...
 
+	LOG_DBG("%s ISFR=0x%08x", dev->config->name, int_status);
+
 	_gpio_fire_callbacks(&data->callbacks, dev, enabled_int);
 
 	/* Clear the port interrupts */
 	config->port_base->ISFR = 0xFFFFFFFF;
+
+	if (data->daisy_dev) {
+		gpio_mcux_port_isr(data->daisy_dev);
```
Comment (agansari) [inline drivers/gpio/gpio_mcux.c]: Haven't seen recursive code that works in a while

**Review by agansari — COMMENTED** [review]
Looks good.
Fixes #13122

File: dts/bindings/gpio/nxp,kinetis-gpio.yaml [hunk 2]
```diff
@@ -25,6 +25,11 @@ properties:
       description: required interrupts
       generation: define
 
+    daisy-gpios:
```
Comment (galak) [inline dts/bindings/gpio/nxp,kinetis-gpio.yaml]: What is this property suppose to mean?

**Comment by jremmert-phytec-iot** [comment]
Tested it with my example that failed the check before, works now.

**Review by dleach02 — COMMENTED** [review]
What are the test cases that this now supports? I do see that it fixes #13122 but curious if there are other samples/tests? 

Otherwise, looks fine.

File: dts/bindings/gpio/nxp,kinetis-gpio.yaml (hunk 2 above)
Comment (dleach02) [reply dts/bindings/gpio/nxp,kinetis-gpio.yaml]: My interpretation is that "daisy" will be the preample for any items that will have a daisy chain of IRQ. In this case, GPIO is daisy chained.

**Pull request closed without merging**